### PR TITLE
Add complex.{__abs__, __eq__, __neg__}

### PR DESCRIPTION
**Pull request by OddCoincidence**
The former required the addition of the [`NotImplemented`](https://docs.python.org/3/library/constants.html#NotImplemented) constant to match cpython`s behavior.

### CODE REVIEW

**Comment by cthulahoops** [comment]
Can we test these using idiomatic python rather than calling the magic methods directly:

Eg:

```
assert abs(complex(3, 4)) == 5
assert complex(3, 4) == complex(3, 4)
```

etc?

**Comment by windelbouwman** [comment]
@cthulahoops I agree, this is cleaner testcode. It will invoke the magic method at some point in time.

File: vm/src/pyobject.rs [hunk 1]
```diff
@@ -130,6 +130,7 @@ pub struct PyContext {
     pub map_type: PyObjectRef,
     pub memoryview_type: PyObjectRef,
     pub none: PyObjectRef,
+    pub not_implemented: PyObjectRef,
```
Comment (windelbouwman) [inline vm/src/pyobject.rs]: Wow, I cannot be believe we did not implement this yet ;)

**Comment by OddCoincidence** [comment]
Thanks to @cthulahoops's suggestion, I think I found a subtle bug in the way the binary magic methods currently are implemented. Filed as #419. I will work around it here, and fix this in another PR.

**Comment by cthulahoops** [comment]
Looks ready to merge, but has conflicts to resolve.

**Comment by windelbouwman** [comment]
I resolved the conflict.

**Pull request merged**